s3/url_test.go

```diff
@@ -0,0 +1,38 @@
+package s3_test
```

Comment:
This is a patterned picked up here: https://medium.com/@matryer/5-simple-tips-and-tricks-for-writing-unit-tests-in-golang-619653f90742

It suggests putting tests in a separate package to hide the internals, and test via the public interface of the package. Happy to discuss and change if we decide on another convention.